packages/lib/src/vis/line/errorBarsGeometry.ts

```diff
@@ -1,62 +1,63 @@
-/* eslint-disable @typescript-eslint/no-non-null-assertion */
-import { type IgnoreValue, type NumArray } from '@h5web/shared/vis-models';
+import { type NumArray } from '@h5web/shared/vis-models';
+import { type BufferAttribute, BufferGeometry } from 'three';
 
-import { type AxisScale } from '../models';
-import H5WebGeometry from '../shared/h5webGeometry';
+import { type H5WebGeometry } from '../models';
 import { createBufferAttr, Z_OUT } from '../utils';
+import { type LineGeometryParams } from './lineGeometry';
 
-interface Params {
-  abscissas: NumArray;
-  ordinates: NumArray;
+interface Params extends LineGeometryParams {
   errors: NumArray;
-  abscissaScale: AxisScale;
-  ordinateScale: AxisScale;
-  ignoreValue?: IgnoreValue;
 }
 
-class ErrorBarsGeometry extends H5WebGeometry<'position', Params> {
-  public constructor(length: number) {
+class ErrorBarsGeometry
+  extends BufferGeometry<Record<'position', BufferAttribute>>
+  implements H5WebGeometry
+{
+  public constructor(private readonly params: Params) {
     super();
+    const { length } = params.ordinates;
     this.setAttribute('position', createBufferAttr(length * 2));
   }
 
-  public update(index: number): void {
+  public update(): void {
```

Comment:
Sorry for the noise in the diff, I renamed `bufferIndex` and destructured `this.attributes.position`... The notable change is of course the `for...of` loop and the use of `continue` instead of `return`.